docs: add construct architecture diagrams

jusiskin opened the pull request:
Fixes #102 

## Summary

Adds architectural diagrams for the primary constructs:

**Core:**

*   `HealthMonitor`
*   `ImportedAcmCertificate`
*   `X509CertificatePem`
*   `X509CertificatePkcs12`

**Deadline:**
   * `ConfigureSpotEventPlugin`
* `Repository`
* `RenderQueue`
* `SpotEventPluginFleet`
* `UsageBasedLicensing`
* `WorkerInstanceFleet`

These are added as draw.io embedded `.svg` files which can be edited using https://app.diagrams.net/. Diagrams are stored the following path convention `packages/aws-rfdk/docs/diagrams/<package>/<ConstructName>.svg` directory of the repository.

The architecture diagrams are embedded in each module's `README.md` and each corresponding constructs' TSDoc header. They are browsable via the README files in GitHub.

The diagrams are also propagated to our API reference documentation in the construct docs pages (both the main docs page and the python-specific one) and the package overviews.

This PR adds documentation that states the expectation for keeping the diagrams updated and the process for doing so. This is linked from the `CONTRIBUTING.md` file.

## Testing

* Built the API reference documentation using a JSII build from this branch
* Confirmed the diagrams appear in the rendered HTML docs for both the main docs and python docs correctly in the following places:
   *    the package overviews (README files)
   *    the construct pages

----

*By submitting this pull request, I confirm that my contribution is made under the terms of the Apache-2.0 license*

jusiskin committed:
docs: add construct architecture diagrams

jusiskin committed:
diagram improvements

jusiskin committed:
fix WorkerInstanceFleet edge label

jusiskin committed:
add docs for referencing arch diagrams from JSDoc headers

horsmand reviewed (commented):
This looks ready for approval for me but we are reviewing the actual diagrams externally to GitHub since the diff here doesn't actually show them. I'll hold off on the approval here until the review of the diagrams is also complete.

jusiskin committed:
address CR feedback

- add missing hitch line in ImportedCertificatePem
- add certificate chain secret to TLS cert constructs
- fix line connections to ACM cert resources

jusiskin committed:
iterations

- reduced extra space in ImportedAcmCertificate
- fixed label placement in X509CertificatePem

jusiskin committed:
test: add test to ensure SVG files have draw.io diagrams embedded

horsmand commented on packages/aws-rfdk/docs/diagrams/tests/svg-embedded-drawio.test.ts:
```diff
@@ -0,0 +1,75 @@
+import * as path from 'path';
+
+/**
+ * This is the regular expression that validates that the dar.io diagram is embedded within the SVG file.
```
```suggestion
 * This is the regular expression that validates that the draw.io diagram is embedded within the SVG file.
```

jusiskin replied:
good catch. fixed

horsmand commented on packages/aws-rfdk/docs/diagrams/tests/svg-embedded-drawio.test.ts:
```diff
@@ -0,0 +1,75 @@
+    });
+
+  });
+});
```
Missing blank line.

jusiskin replied:
fixed

horsmand commented on packages/aws-rfdk/docs/diagrams/tests/svg-embedded-drawio.test.ts:
```diff
@@ -0,0 +1,75 @@
+    const stat = fs.statSync(listingPath);
+    if (stat.isDirectory()) {
+      const dirDiagrams = findSvgDiagrams(listingPath);
+      diagrams.splice(diagrams.length, 0, ...dirDiagrams);
```
This is fine by me to leave as-is, but I'm curious why you'd use `splice` over `concat`?

jusiskin replied:
no reason. concat is more readable, so I switched it over

horsmand replied:
I think `concat` works without needing the `...` but I don't think it makes a difference.

jusiskin committed:
chore: address CR feedback for draw.io SVG tests

jusiskin committed:
docs: clarify label in X509CertificatePem arch diagram

ddneilson reviewed (approved):
This is great, thanks Josh! The added test is a great idea!
